### PR TITLE
Cache package sizes

### PR DESCRIPTION
**Pull request by wgoodall01**
(Fixes issue #1)

This PR implements caching for the module sizes.
Every time `getSizes()` is run and calculates the size of a bundle, it will write the size information to a JSON file located wherever `package-size` is installed. Then, for subsequent runs of `getSizes()`, the cached values will be read from disk and used instead of rebuilding the package. To disable the cache and force a module to be rebuilt, pass `--no-cache` to the CLI or `{cache:false}` in the `options` obj. To clear the cache, run `package-size clear-cache`.

### CODE REVIEW

File: lib/cache.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+const fs = require('fs-extra');
+const path = require('path');
+const co = require('co');
+
+const cacheFilePath = path.join(__dirname, '__cache.json');
```
Comment (egoist) [inline lib/cache.js]: hmm, we should not store cache in the package directory. simply use something like https://github.com/sindresorhus/conf

File: lib/index.js [hunk 2]
```diff
@@ -196,14 +203,19 @@ module.exports = co.wrap(function*(name, opts = {}) {
...
     name,
     size: asset.size - 2753, // minus webpack runtime size
     minified: find(prodAssets, v => {
       return decodeURIComponent(v.name) === `prod.js`
     }).size - 537,
     gzipped: prodGzipSize - 297
   }
+
+  // Cache the package size information.
+  yield cache.set(name, info);
```
Comment (egoist) [inline lib/index.js]: maybe we can use `name + version` as the key to store cache.

**Comment by wgoodall01** [comment]
Implemented your changes --

For each set of modules, after they are fetched with `yarn`, a cache key is generated from the names and exact versions -- this is then used to store the sizes with the `conf` module.

**Comment by egoist** [comment]
Thanks for the PR, released in `v0.15`

**Pull request merged**